### ISSUE TITLE
使用自己的数据集训练yolov3不能收敛， mAP不上升

### ISSUE THREAD

**Issue by henryccl** [open-mmlab/mmdetection#7026]
我在使用mmdetection中的yolo v3训练自己的数据集（VOC格式）时，发现训练了好几百个epoch mAP都不上升，随后我用同样的配置文件训练了balloon数据集发现可以正常收敛，我想应该是我数据集的配置问题，这种使用yolov3训练新的数据集的情况下，需要注意些什么呢？我该从哪出发才能找到问题呢？

**Comment by henryccl** [comment]
配置文件如下
`# 这个新的配置文件继承自一个原始配置文件，只需要突出必要的修改部分即可
_base_ = '../yolo/yolov3_d53_mstrain-608_273e_coco.py'
# python tools/train.py configs/xray_detect/xray_yolo_voc.py --resume_from='work_dirs/xray_yolov3/epoch_99.pth' --validate=20
# 修改数据集相关设置
classes = ("knife", "umbrella", "expandableBaton", "electronicEquipment", "battery", "sharpTools",
           "plasticBottleWithaNozzle", "plasticBeverageBottle", "smallGlassBottle", "seal", "scissors", "electricBaton")

# model settings
model = dict(
    type='YOLOV3',
    backbone=dict(
        type='Darknet',
        depth=53,
        out_indices=(3, 4, 5),
        init_cfg=dict(type='Pretrained', checkpoint='open-mmlab://darknet53')),
    neck=dict(
        type='YOLOV3Neck',
        num_scales=3,
        in_channels=[1024, 512, 256],
        out_channels=[512, 256, 128]),
    bbox_head=dict(
        type='YOLOV3Head',
        num_classes=12,
        in_channels=[512, 256, 128],
        out_channels=[1024, 512, 256],
        anchor_generator=dict(
            type='YOLOAnchorGenerator',
            base_sizes=[[(116, 90), (156, 198), (373, 326)],
                        [(30, 61), (62, 45), (59, 119)],
                        [(10, 13), (16, 30), (33, 23)]],
            strides=[32, 16, 8]),
        bbox_coder=dict(type='YOLOBBoxCoder'),
        featmap_strides=[32, 16, 8],
        loss_cls=dict(
            type='CrossEntropyLoss',
            use_sigmoid=True,
            loss_weight=1.0,
            reduction='sum'),
        loss_conf=dict(
            type='CrossEntropyLoss',
            use_sigmoid=True,
            loss_weight=1.0,
            reduction='sum'),
        loss_xy=dict(
            type='CrossEntropyLoss',
            use_sigmoid=True,
            loss_weight=2.0,
            reduction='sum'),
        loss_wh=dict(type='MSELoss', loss_weight=2.0, reduction='sum')),
    # training and testing settings
    train_cfg=dict(
        assigner=dict(
            type='GridAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.5,
            min_pos_iou=0)),
    test_cfg=dict(
        nms_pre=1000,
        min_bbox_size=0,
        score_thr=0.05,
        conf_thr=0.005,
        nms=dict(type='nms', iou_threshold=0.45),
        max_per_img=100))
# dataset settings
dataset_type = 'VOCDataset'
data_root = '/home/user301/usr/Henry/Xray_dataset_VOC2007/'
img_norm_cfg = dict(mean=[0, 0, 0], std=[255., 255., 255.], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile', to_float32=True),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(
        type='Expand',
        mean=img_norm_cfg['mean'],
        to_rgb=img_norm_cfg['to_rgb'],
        ratio_range=(1, 2)),
    dict(
        type='MinIoURandomCrop',
        min_ious=(0.4, 0.5, 0.6, 0.7, 0.8, 0.9),
        min_crop_size=0.3),
    dict(type='Resize', img_scale=[(320, 320), (608, 608)], keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(608, 608),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=4,
    train=dict(
        type=dataset_type,
        classes=classes,
        ann_file=data_root + 'train.txt',
        img_prefix=data_root + '',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        classes=classes,
        ann_file=data_root + 'val.txt',
        img_prefix=data_root + '',
        pipeline=test_pipeline),
    test=dict(
        type=dataset_type,
        classes=classes,
        ann_file=data_root + 'val.txt',
        img_prefix=data_root + '',
        pipeline=test_pipeline))
# optimizer
optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=0.0005)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=2000,  # same as burn-in in darknet
    warmup_ratio=0.1,
    step=[218, 246])
# runtime settings
checkpoint_config = dict(interval=40)
runner = dict(type='EpochBasedRunner', max_epochs=400)
evaluation = dict(interval=20, metric=['mAP'])`

**Comment by sunnyisabaster** [comment]
I think u should check ur dataset files, not only config files.

**Comment by RangiLyu** [comment]
batch size is too small and learning rate is too large

**Comment by Luxiangxiang21** [comment]
I'm having the same problem, but I've changed my learning rate and it's still not going up.

**Comment by silverlining21** [comment]
> I'm having the same problem, but I've changed my learning rate and it's still not going up.

same problem~  adjust lr and batchsize accordingly, still got the same result `dets=0 recall=0 ap=0`
